src/cocotb/types/logic_array.py

```diff
@@ -320,10 +339,23 @@ def _get_str(self) -> str:
                 )
         return self._value_as_str
 
-    def _get_int(self) -> int:
+    def _get_int(
+        self,
+        resolve: "Literal['error'] | Literal['low'] | Literal['high'] | Literal['random']",
+    ) -> int:
         if self._value_as_int is None:
             # May convert list to str before converting to int.
-            self._value_as_int = int(self._get_str(), 2)
+            try:
+                self._value_as_int = int(self._get_str(), 2)
+            except ValueError as e:
+                if resolve == "error":
+                    raise e
+
+                self._value_as_int = int(
+                    RE_UNRESOLVED.sub(RESOLVE_MAP[resolve], self._get_str()),
+                    2,  # type: ignore
+                )
```

Comment:
Can't save it. What happens if they get the integer value again and change the resolve method?